ext/opentelemetry-exporter-cloud-trace/tests/test_cloud_trace_propagator.py

```diff
@@ -0,0 +1,123 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import typing
+import unittest
+
+import opentelemetry.trace as trace
+from opentelemetry.context import get_current
+from opentelemetry.exporter.cloud_trace.cloud_trace_propagator import (
+    _TRACE_CONTEXT_HEADER_NAME,
+    CloudTraceFormatPropagator,
+)
+from opentelemetry.trace.span import (
+    SpanContext,
+    TraceFlags,
+    get_hexadecimal_trace_id,
+)
+
+
+def get_dict_value(dict_object: typing.Dict[str, str], key: str) -> str:
```

Comment:
Do we have any conventions for typing hints? I see typing hints being using in a few places, but I don't see much consistency.

Reply:
The tl;dr is that these hints never get evaluated, and don't follow any convention in this repo. They're for the benefit of the person reading the code only and can be safely removed.

Longer answer:

Every public method in `opentelemetry-api` should have type hints. Tests in that package may have type hints, but the point is to uncover unexpected type changes in the API, not to document the tests.

`tox -e mypy` runs:

```
mypy --namespace-packages opentelemetry-api/src/opentelemetry/
mypy --namespace-packages --config-file=mypy-relaxed.ini opentelemetry-api/tests/
```

We don't run `mypy` for `opentelemetry-sdk`, but the public-facing parts of the SDK (including all the exporter classes) should have type hints, and we really _should_ have a tox target that checks this.

We also don't run `mypy` on ext packages, so the type hints here are just documentation. If we wanted type checking for ext packages similar to the API package we could add a target that ran:

```
mypy --namespace-packages --config-file=mypy-relaxed.ini ext/opentelemetry-exporter-cloud-trace/tests
```

...but this fails right now, and we may need to [add type stubs](https://mypy.readthedocs.io/en/latest/installed_packages.html#making-pep-561-compatible-packages) for dependencies to get meaningful results from `mypy` here.

Reply:
Issue for tox not running mypy on the sdk: #773